adapta/utils/decorators/_logging.py

```diff
@@ -0,0 +1,51 @@
+""" Module for common decorator methods. """
+from functools import wraps
+
+from adapta.logs import SemanticLogger
+from adapta.metrics import MetricsProvider
+from adapta.utils._common import operation_time
+
+
+def run_time_metrics(method_type: str, pass_logger: bool = False, pass_metrics_provider: bool = False):
```

Comment:
Can we call it `metric_name` instead of `method_type`?

Reply:
Renamed.